src/serac/coefficients/coefficient_extensions.hpp

```diff
@@ -14,41 +14,170 @@
+struct eval_t<T, std::enable_if_t<std::is_base_of_v<mfem::VectorCoefficient, T>>> {
+  using type = mfem::Vector;
+};
+
+/// methods for evaluating coefficient stuff with the same prototype
+double eval(double& d, mfem::ElementTransformation&, const mfem::IntegrationPoint&) { return d; }
+
+/// evaluates an mfem::Coefficient
+double eval(mfem::Coefficient& c, mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip)
+{
+  return c.Eval(Tr, ip);
+}
+
+/// evaluates a mfem::VectorCoefficient
+mfem::Vector eval(mfem::VectorCoefficient& v, mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip)
+{
+  mfem::Vector temp(v.GetVDim());
+  v.Eval(temp, Tr, ip);
+  return temp;
+}
+
+
+template <typename T, typename = void>
+struct is_iterable : std::false_type {};
+  
+template <typename T>
+struct is_iterable<T, std::void_t<decltype(std::declval<T>().begin()),
+                                  decltype(std::declval<T>().end())>>
+    : std::true_type {};
+
+template <typename T, typename = void>
+struct has_brackets : std::false_type {};
+  
+template <typename T>
+struct has_brackets<T, std::void_t<decltype(std::declval<T>()[0])>>
+    : std::true_type {};
+  
+}  // namespace detail
+
+namespace digitize {
 /**
- * @brief MakeTrueEssList takes in a FESpace, a vector coefficient, and produces a list
- *  of essential boundary conditions
+ * @brief takes floating point value and rounds down to the nearest integer
+ * @param[in] v floating point value
+ */
+
+[[maybe_unused]] static int floor(double v) { return static_cast<int>(v); }
+
+/**
+ * @brief Thresholds a real function so that it is 2 if greater than 0.
+ * @param[in] v floating point value
+ */
+[[maybe_unused]] static int threshold(double v) { return v > 0. ? 2 : 1; }
+
+/**
+ * @brief Checks if floating point value is equal to 1, if return 2 otherwise return 1.
+ */
+[[maybe_unused]] static int equals1(double v) { return v == 1. ? 2 : 1; }
+}  // namespace digitize
+
+/**
+ * @brief MakeEssList takes in a FESpace, a vector coefficient, and produces a list
+ * of essential boundary conditions
  *
  * @param[in] pfes A finite element space for the constrained grid function
- * @param[in] c A VectorCoefficient that is projected on to the mesh. All
+ * @param[in] c A coefficient that is projected on to the mesh. All
  * d.o.f's are examined and those that are the condition (> 0.) are appended to
  * the vdof list.
- * @return The list of true dofs that should be part of the essential boundary conditions
+ * @return The list of vector dofs that should be
+ * part of the essential boundary conditions
  */
-mfem::Array<int> MakeTrueEssList(mfem::ParFiniteElementSpace& pfes, mfem::VectorCoefficient& c);
+  template <typename T, typename SFINAE = std::enable_if_t<std::is_base_of_v<mfem::Coefficient, T> ||
+							   std::is_base_of_v<mfem::VectorCoefficient, T>>>
+mfem::Array<int> MakeEssList(mfem::ParFiniteElementSpace& pfes, T& c)
+{
+  mfem::Array<int> ess_vdof_list(0);
```

Comment:
Minor: Would this be more readable if it were just default-constructed?

Also, unlike most `std::vector` implementations, MFEM resizing here is naive (resizes to size + 1 instead of e.g. size * 2) and will reallocate + copy after every `Append`. Would it be worth considering making a single conservative allocation of `pfes.GetVSize()` here and then shrinking at the end?

Reply:
it seems like it seems like `mfem::Array` resizing is not as naive as that (thankfully):

```cpp
template <class T>
inline void Array<T>::GrowSize(int minsize) {
  const int nsize = std::max(minsize, 2 * data.Capacity());
  Memory<T> p(nsize, data.GetMemoryType());
  p.CopyFrom(data, size);
  p.UseDevice(data.UseDevice());
  data.Delete();
  data = p;
}
```

so it shouldn't be too bad to keep as is. Alternatively, maybe just reserve something like 1/8 of the `pfes.GetVSize()` before the loop.

Reply:
Thanks for the correction - I think maybe I was thinking of the `mfem::Vector` resize logic?

Reply:
What do you think about using `std::vector` and then just converting to `mfem::Array` at the end?

Reply:
Should I just move to "default constructed" then? It seems like if I use `std::vector` we'll have an extra copy. At the end of the day, this information probably needs to be `mfem::Array<int>`.